### PR TITLE
RAB-1823: Support new attribute group sheet & column in templates

### PR DESCRIPTION
**Pull request by marinfazioakeneo**
In this PR we handle the new sheet `attribute_groups` & the new column `attribute_group` in each family.
This new attribute group column is also added to each json family template as a new attribute property.

### CODE REVIEW

File: src/Command/LintTemplatesCommand.php [hunk 1]
```diff
@@ -251,6 +262,10 @@ private function lintFamilies(array $families, array $familyCodesInIndustries):
                             choices: AttributeType::getChoices(),
                             message: 'This value is not a valid attribute type.',
                         ),
+                        'group' => new Choice(
+                            choices: $attributeGroupsCodes,
+                            message: 'This value is not a valid attribute group.',
```
Comment (StevenVAIDIE) [inline src/Command/LintTemplatesCommand.php:267]:
```suggestion
                            message: 'The attribute group does not exist.',
```

File: src/Command/MinifyTemplatesCommand.php [hunk 2]
```diff
@@ -104,4 +113,23 @@ private function addAttributeOptionsToFamilyTemplate(array $familyTemplate, arra
...
+
+    private function addAttributeGroupsToFamilyTemplates(array $familyTemplates, array $attributeGroups): array
+    {
+        return array_map(
+            fn (array $familyTemplate) => $this->addAttributeGroupsToFamilyTemplate($familyTemplate, $attributeGroups),
+            $familyTemplates,
+        );
+    }
+
+    private function addAttributeGroupsToFamilyTemplate(array $familyTemplate, array $attributeGroups): array
```
Comment (StevenVAIDIE) [inline src/Command/MinifyTemplatesCommand.php:125]: For me attribute group should not de defined at each attribute level but at family level like the attribute options

**Review by antonstaradzinau — APPROVED** [review]

File: src/Command/MinifyTemplatesCommand.php [hunk 3]
```diff
@@ -104,4 +113,17 @@ private function addAttributeOptionsToFamilyTemplate(array $familyTemplate, arra
 
         return $familyTemplate;
     }
+
+    private function addAttributeGroupsToFamilyTemplates(array $familyTemplates, array $attributeGroups): array
+    {
+        return array_map(function (array $familyTemplate) use ($attributeGroups) {
+            $attributeGroupCodes = array_map(fn ($attribute) => $attribute['group'], $familyTemplate['attributes']);
```
Comment (StevenVAIDIE) [inline src/Command/MinifyTemplatesCommand.php:120]:
```suggestion
            $attributeGroupCodesInFamilyTemplate = array_map(static fn ($attribute) => $attribute['group'], $familyTemplate['attributes']);
```

**Review by StevenVAIDIE — APPROVED** [review]

File: dist/minified.json [hunk 4]
```diff
@@ -1 +1 @@
..._kirin","labels":{"en_US":"Huawei Kirin"},"attribute":"processor"},"processor_intel_atom":{"code":"processor_intel_atom","labels":{"en_US":"Intel Atom"},"attribute":"processor"},"processor_hisilicon_kirin":{"code":"processor_hisilicon_kirin","labels":{"en_US":"HiSilicon Kirin"},"attribute":"processor"},"processor_nvidia_tegra":{"code":"processor_nvidia_tegra","labels":{"en_US":"NVIDIA Tegra"},"attribute":"processor"},"processor_amd_ryzen_mobile":{"code":"processor_amd_ryzen_mobile","labels":{"en_US":"AMD Ryzen Mobile"},"attribute":"processor"},"processor_mediatek_helio":{"code":"processor_mediatek_helio","labels":{"en_US":"MediaTek Helio"},"attribute":"processor"},"processor_dual_core":{"code":"processor_dual_core","labels":{"en_US":"Dual Core"},"attribute":"processor"},"processor_mediatek":{"code":"processor_mediatek","labels":{"en_US":"MediaTek"},"attribute":"processor"}}},"snacks":{"code":"snacks","labels":{"en_US":"Snacks"},"description":{"en_US":"Satisfy your cravings with our 'Snacks' family template - a delicious assortment of pre-built attributes to effortlessly manage your favorite treats, making snack time a delightful experience."},"attribute_as_main_media":"main_image",
```
Comment (ghoussard) [inline dist/minified.json:1]: Please don't commit it, there's a workflow which is triggered to do it on merge on min 👍🏻

File: templates/attribute_groups.json [hunk 5]
```diff
@@ -0,0 +1,7 @@
+{
+  "other": { "code": "other", "labels": { "en_US": "Other" } },
+  "dpp": {
+    "code": "dpp",
+    "labels": { "en_US": "DPP (Digital Product Passport)" }
+  }
```
Comment (ghoussard) [inline templates/attribute_groups.json:6]: Prettier is ok with this formatting ?

File: tests/Command/GenerateTemplatesCommand/GenerateTemplatesCommandTest.php [hunk 6]
```diff
@@ -49,7 +49,7 @@ private function assertDirectoryEquals(string $expectedDirectoryPath, string $ac
             $expectedFilePath = sprintf('%s/%s', $expectedDirectoryPath, $expectedFile);
 
             if (is_dir($expectedFilePath)) {
-                $this->assertDirectoryEquals($actualFilePath, $expectedFilePath);
+                $this->assertDirectoryEquals($expectedFilePath, $actualFilePath);
```
Comment (ghoussard) [inline tests/Command/GenerateTemplatesCommand/GenerateTemplatesCommandTest.php:52]: Weird we didn't see this issue earlier 🤔

File: tests/Command/LintTemplatesCommand/LintTemplatesCommandTest.php [hunk 7]
```diff
@@ -89,6 +89,8 @@ public static function provideInvalidTemplates(): iterable
                 '[families][missing_attribute_negative_allowed_family][attributes][1][negative_allowed] This field is missing.',
                 '[families][wrong_attribute_validation_rule_family][attributes][3][validation_rule] This value is not a valid validation rule. Please use one of the following : url,email.',
                 '[families][missing_attribute_validation_rule_family][attributes][1][validation_rule] This field is missing.',
+                '[families][missing_attribute_group_family][attributes][1][group] This field is missing.',
+                '[families][wrong_attribute_group_family][attributes][1][group] This attribute group does not exist.'
             ]
         ];
         yield 'Invalid attribute options' => [
```
Comment (ghoussard) [inline tests/Command/LintTemplatesCommand/LintTemplatesCommandTest.php:96]: It misses all tests case about attribute groups file validation. You can take example on the industry test case line 37, it is almost the same tests cases

**Review by ghoussard — APPROVED** [review]

**Pull request merged**